specs/WebNotification.md

```diff
@@ -0,0 +1,610 @@
+        BOOL userInitiated = FALSE;
+        wil::unique_cotaskmem_string uri;
+        CHECK_FAILURE(args->get_PermissionKind(&kind));
+        CHECK_FAILURE(args->get_IsUserInitiated(&userInitiated));
+        CHECK_FAILURE(args->get_Uri(&uri));
+
+        COREWEBVIEW2_PERMISSION_STATE state;
+
+        auto cached_key = std::make_tuple(std::wstring(uri.get()), kind, userInitiated);
+        auto cached_permission = m_cached_permissions.find(cached_key);
+        if (cached_permission != m_cached_permissions.end())
+        {
+            state =
+                (cached_permission->second ? COREWEBVIEW2_PERMISSION_STATE_ALLOW
+                                           : COREWEBVIEW2_PERMISSION_STATE_DENY);
+        }
+        else
+        {
+            std::wstring message = L"An iframe has requested device permission for ";
+            message += SettingsComponent::NameOfPermissionKind(kind);
+            message += L" to the website at ";
+            message += uri.get();
+            message += L"?\n\n";
+            message += L"Do you want to grant permission?\n";
+            message +=
+                (userInitiated ? L"This request came from a user gesture."
+                               : L"This request did not come from a user gesture.");
+
+            int response = MessageBox(
+                nullptr, message.c_str(), L"Permission Request",
+                MB_YESNOCANCEL | MB_ICONWARNING);
+            switch (response)
+            {
+            case IDYES:
+                m_cached_permissions[cached_key] = true;
+                state = COREWEBVIEW2_PERMISSION_STATE_ALLOW;
+                break;
+            case IDNO:
+                m_cached_permissions[cached_key] = false;
+                state = COREWEBVIEW2_PERMISSION_STATE_DENY;
+                break;
+            default:
+                state = COREWEBVIEW2_PERMISSION_STATE_DEFAULT;
+                break;
+            }
+        }
+
+        CHECK_FAILURE(args->put_State(state));
+        CHECK_FAILURE(deferral->Complete());
+    });
+    return S_OK;
+}
+
+```
+
+## Filter Notifications from a specific doamin and send local toast
+
+Learn more about sending a local toast [Send a local toast notification from a C# app - Windows apps | Microsoft Learn](https://learn.microsoft.com/windows/apps/design/shell/tiles-and-notifications/send-local-toast?tabs=uwp).
+
+### C# Sample
+```csharp
+using Microsoft.Toolkit.Uwp.Notifications;
+...
+void WebView_NotificationReceived(object sender, CoreWebView2NotificationReceivedEventArgs args)
+{
+    CoreWebView2Deferral deferral = args.GetDeferral();
+    var notification = args.Notification;
+
+    // Show notification with local MessageBox
+    string message = "Received notification from " + args.Uri + " with body " + notification.Body;
+    MessageBox.Show(message);
+
+    // Requires Microsoft.Toolkit.Uwp.Notifications NuGet package version 7.0 or greater
+    var toastContent = new ToastContentBuilder()
```

Comment:
The C++ & C# samples should match. The C++ one is doing filtering and the C# isn't. The C# one is showing a message box and using ToastContentBuilder and C++ is only doing messagebox. It would be nice to have both do filtering and both do fancy display of toasts. If we don't see an obvious way of doing a fancy toast for C++ then use ToastContentBuilder for C# And messagebox for C++ with a comment explaining in each